### PR TITLE
Added new cmdlets for flowLog resource

### PR DESCRIPTION
**Pull request by irrogozh**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_) https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/485
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
Design: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/485

File: src/Network/Network/NetworkWatcher/FlowLog/SetAzNetworkWatcherFlowLogCommand.cs [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+        {
+            ResourceIdentifier flowLogInfo = new ResourceIdentifier(id);
+
+            this.Name = flowLogInfo.ResourceName;
+            this.ResourceGroupName = flowLogInfo.ResourceGroupName;
+
+            string parent = flowLogInfo.ParentResource;
+            string[] tokens = parent.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+
+            this.NetworkWatcherName = tokens[1];
```
Comment (wyunchi-ms) [inline src/Network/Network/NetworkWatcher/FlowLog/SetAzNetworkWatcherFlowLogCommand.cs]: What if tokens length is 0?

**Comment by wyunchi-ms** [comment]
Hi @irrogozh , could you please also add test case and help files for these cmdlets.

File: src/Network/Network/NetworkWatcher/FlowLog/SetAzNetworkWatcherFlowLogCommand.cs (hunk 1 above)
Comment (irrogozh) [reply src/Network/Network/NetworkWatcher/FlowLog/SetAzNetworkWatcherFlowLogCommand.cs]: add validation

File: src/Network/Network/NetworkWatcher/FlowLog/NewAzNetworkWatcherFlowLogCommand.cs [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+            this.FlowLogs.CreateOrUpdate(this.ResourceGroupName, this.NetworkWatcherName, this.Name, flowLogParameters);
+            MNM.FlowLog flowLogResult = this.FlowLogs.Get(this.ResourceGroupName, this.NetworkWatcherName, this.Name);
+
+            return NetworkResourceManagerProfile.Mapper.Map<PSFlowLogResource>(flowLogResult);
+        }
+
+        private void ValidateInput()
+        {
+            string[] splittedTargetResourceId = this.TargetResourceId.Split('/');
+            if (splittedTargetResourceId.Count() < 9 || string.IsNullOrEmpty(splittedTargetResourceId[7]) || !splittedTargetResourceId[7].Equals("networkSecurityGroups", StringComparison.OrdinalIgnoreCase))
```
[the remark this answers is not in the document]
Comment (irrogozh) [reply src/Network/Network/NetworkWatcher/FlowLog/NewAzNetworkWatcherFlowLogCommand.cs]: done

File: src/Network/Network/NetworkWatcher/FlowLog/GetAzNetworkWatcherFlowLogCommand.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            ParameterSetName = "SetByLocation")]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string Name { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (ParameterSetName.Contains("SetByResourceId"))
```
[the remark this answers is not in the document]
Comment (irrogozh) [reply src/Network/Network/NetworkWatcher/FlowLog/GetAzNetworkWatcherFlowLogCommand.cs]: done

File: src/Network/Network/NetworkWatcher/FlowLog/SetAzNetworkWatcherFlowLogCommand.cs [hunk 4]
```diff
@@ -0,0 +1,671 @@
...
+
+            MNM.FlowLog flowLogParameters = GetFlowLogParametersFromRequest();
+
+            this.FlowLogs.CreateOrUpdate(this.ResourceGroupName, this.NetworkWatcherName, this.Name, flowLogParameters);
+            MNM.FlowLog flowLogResult = this.FlowLogs.Get(this.ResourceGroupName, this.NetworkWatcherName, this.Name);
+
+            return NetworkResourceManagerProfile.Mapper.Map<PSFlowLogResource>(flowLogResult);
+        }
+
+        private void ValidateInput()
```
[the remark this answers is not in the document]
Comment (irrogozh) [reply src/Network/Network/NetworkWatcher/FlowLog/SetAzNetworkWatcherFlowLogCommand.cs]: done

**Pull request merged**